clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemResourceDefinition.java

```diff
@@ -98,7 +98,7 @@ public void register(SubsystemRegistration parentRegistration) {
 
         ResourceDescriptor descriptor = new ResourceDescriptor(this.getResourceDescriptionResolver())
                 .addCapabilities(LOCAL_CLUSTERING_CAPABILITIES.values())
-                .addCapabilities(CLUSTERING_CAPABILITIES.values())
+                .addCapabilities((context, model) -> context.getRootResourceRegistration().getSubModel(context.getCurrentAddress().getParent().append(PathElement.pathElement(ModelDescriptionConstants.SUBSYSTEM, "jgroups"))) == null, CLUSTERING_CAPABILITIES.values())
```

Comment:
@pferraro I assume this check runs during Stage.MODEL during boot. That's not reliable. Subsystems are added concurrently and there is no way for one to know whether a different subsystem will shortly be added by a different thread.

Even ignoring concurrency there's no requirement that jgroups is added before infinispan, so the test would have to be deferred by adding a step. But the concurrency aspect is the real blocker here.

Also, in this case I think duplicate registration is actually valid. If a user has both jgroups and infinispan and they remove jgroups, their configuration would still provide this capability. If jgroups provides the cap via a resource that requires reload if removed, then the user can remove jgroups, reload, and still have this cap. Without dual registration, the remove will fail due to the cap no longer being present.